src/expandableCalendar/Context/Provider.tsx

```diff
@@ -34,158 +43,132 @@ interface Props {
-
-    if (shouldAnimateTodayButton(this.props)) {
-      const animationData = getPositionAnimation(date, this.props.todayBottomMargin);
+  const animateTodayButton = (date: string) => {
+    if (shouldAnimateTodayButton(props)) {
+      const animationData = getPositionAnimation(date, todayBottomMargin);
 
-      Animated.spring(this.state.buttonY, {
+      Animated.spring(buttonY.current, {
         ...animationData
       }).start();
     }
-  }
-
-  animateOpacity(disabled: boolean) {
-    const {shouldAnimateOpacity, getOpacityAnimation} = this.presenter;
+  };
 
-    if (shouldAnimateOpacity(this.props)) {
-      const animationData = getOpacityAnimation(this.props, disabled);
+  const animateOpacity = (disabled: boolean) => {
+    if (shouldAnimateOpacity(props)) {
+      const animationData = getOpacityAnimation(props, disabled);
 
-      Animated.timing(this.state.opacity, {
+      Animated.timing(opacity.current, {
         ...animationData
       }).start();
     }
-  }
-
-  onTodayPress = () => {
-    const today = this.presenter.getTodayDate();
-    this.setDate(today, UpdateSources.TODAY_PRESS);
   };
 
-  renderTodayButton() {
-    const {disabled, opacity, buttonY, buttonIcon} = this.state;
-    const {getTodayFormatted} = this.presenter;
+  const onTodayPress = useCallback(() => {
+    _setDate(getTodayDate(), UpdateSources.TODAY_PRESS);
+  }, [_setDate]);
+
+  const renderTodayButton = () => {
     const today = getTodayFormatted();
 
     return (
-      <Animated.View style={[this.style.todayButtonContainer, {transform: [{translateY: buttonY}]}]}>
+      <Animated.View style={[style.current.todayButtonContainer, {transform: [{translateY: buttonY.current}]}]}>
         <TouchableOpacity
-          style={[this.style.todayButton, this.props.todayButtonStyle]}
-          onPress={this.onTodayPress}
-          disabled={disabled}
+          style={[style.current.todayButton, todayButtonStyle]}
+          onPress={onTodayPress}
+          disabled={isDisabled}
         >
-          <Animated.Image style={[this.style.todayButtonImage, {opacity}]} source={buttonIcon} />
-          <Animated.Text allowFontScaling={false} style={[this.style.todayButtonText, {opacity}]}>
+          <Animated.Image style={[style.current.todayButtonImage, {opacity: opacity.current}]} source={buttonIcon}/>
+          <Animated.Text allowFontScaling={false} style={[style.current.todayButtonText, {opacity: opacity.current}]}>
             {today}
           </Animated.Text>
         </TouchableOpacity>
       </Animated.View>
     );
-  }
+  };
 
-  render() {
-    return (
-      <CalendarContext.Provider value={this.getProviderContextValue()}>
-        <View style={[styles.container, this.props.style]}>{this.props.children}</View>
-        {this.props.showTodayButton && this.renderTodayButton()}
-      </CalendarContext.Provider>
-    );
-  }
-}
+  return (
+    <CalendarContext.Provider value={getProviderContextValue()}>
```

Comment:
By invoking `getProviderContextValue` each render you create a new object and a new ref

Reply:
Moved to useMemo